dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2127,166 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _get_null_and_sample_count(self):
+        """
+        Gets the null_count and sample_size of each column.
+
+        :return: Dictionary where column id maps to a list of [null_count, sample_size]
+        :rtype: dict
+        """
+        counts = {}
+        for col_id in range(len(self._profile)):
+            null_count = getattr(self._profile[col_id], "null_count", 0)
+            sample_size = getattr(self._profile[col_id], "sample_size", 0)
+            counts[col_id] = [null_count, sample_size]
+        return counts
+
+    def _calc_nan_replication_metrics(self, clean_samples, prev_counts):
+        """
+        Calculate metrics needed for replicating NaN values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        :param prev_counts: null_count and sample_size of each column before update
+        :type prev_counts: dict
+        """
+        if not self.options.synthetic_data.replicate_nan.is_enabled:
+            return
+
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            null_count = getattr(self._profile[col_id], "null_count")
```

Comment:
add something like `compiler = self._profile[id]` so you don't have to keep repeating the `self._profile[col_id]` call